source/agora/common/TransactionPool.d

```diff
@@ -193,32 +194,17 @@ public class TransactionPool
 
     /***************************************************************************
 
-        Walk over the transactions in the pool and call the provided delegate
-        with each hash and transaction
-
-        Params:
-            dg = the delegate to call
-
         Returns:
-            the status code of the delegate, or zero
+            an input range of (hash, transaction) tuples
 
     ***************************************************************************/
 
-    public int opApply (scope int delegate(Hash hash, Transaction tx) dg) @trusted
+    public auto opSlice () @trusted
```

Comment:
If the resulting range is not safe, isn't it better to not marked this as `@trusted` ?

Reply:
Well the method itself is trusted, but the returned type is not. I wish I could do `@trusted(auto)` hah.